### ISSUE TITLE
Rewrite POM according to canonical ordering and introduce Tidy Maven Plugin

### ISSUE THREAD

**Issue by mkordas** [checkstyle/checkstyle#809]
Right now Checkstyle's POM do not follow [canonical ordering](http://maven.apache.org/developers/conventions/code.html#POM_Code_Convention).

There is nice plugin to make sure that POM has correct order of entries. More info [here](http://mojo.codehaus.org/tidy-maven-plugin/index.html).

It can be incorporated into build by adding
```xml
<plugin>
  <groupId>org.codehaus.mojo</groupId>
  <artifactId>tidy-maven-plugin</artifactId>
  <version>1.0-beta-1</version>
  <executions>
    <execution>
      <id>validate</id>
      <phase>validate</phase>
      <goals>
        <goal>check</goal>
      </goals>
    </execution>
  </executions>
</plugin>
```

**Comment by romani** [comment]
please provide changes to POM.

**Comment by romani** [comment]
resolved , will be in 6.5
